### PR TITLE
Have a single SyntaxNavigator object that is shared between C# and VB.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-compiler

File: src/Compilers/Core/Portable/Syntax/SyntaxNavigator.cs [hunk 1]
```diff
@@ -7,11 +7,39 @@
...
+            /* 001 */ t =>                                             t.IsDocumentationCommentTrivia,
+            /* 010 */ t =>                            t.IsDirective,
+            /* 011 */ t =>                            t.IsDirective || t.IsDocumentationCommentTrivia,
+            /* 100 */ t => t.IsSkippedTokensTrivia,
+            /* 101 */ t => t.IsSkippedTokensTrivia                  || t.IsDocumentationCommentTrivia,
+            /* 110 */ t => t.IsSkippedTokensTrivia || t.IsDirective,
+            /* 111 */ t => t.IsSkippedTokensTrivia || t.IsDirective || t.IsDocumentationCommentTrivia,
+        };
+
+        private Func<SyntaxTrivia, bool> GetStepIntoFunction(bool skipped, bool directives, bool docComments)
```
Comment (cston) [inline src/Compilers/Core/Portable/Syntax/SyntaxNavigator.cs]: `static`

File: src/Compilers/Core/Portable/Syntax/SyntaxNavigator.cs [hunk 2]
```diff
@@ -7,11 +7,39 @@
 
 namespace Microsoft.CodeAnalysis
 {
-    internal abstract class AbstractSyntaxNavigator
+    internal class SyntaxNavigator
```
Comment (cston) [inline src/Compilers/Core/Portable/Syntax/SyntaxNavigator.cs]: `sealed`

File: src/Compilers/Core/Portable/Syntax/SyntaxNavigator.cs [hunk 3]
```diff
@@ -7,11 +7,39 @@
...
     {
         private const int None = 0;
 
-        protected abstract Func<SyntaxTrivia, bool> GetStepIntoFunction(bool skipped, bool directives, bool docComments);
+        public static readonly SyntaxNavigator Instance = new SyntaxNavigator();
```
Comment (cston) [inline src/Compilers/Core/Portable/Syntax/SyntaxNavigator.cs]: Consider making constructor `private`.

File: src/Dependencies/PooledObjects/Microsoft.CodeAnalysis.PooledObjects.shproj [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup Label="Globals">
-    <ProjectGuid>cd77a8cc-2885-47a7-b99c-fbf13353400b</ProjectGuid>
+    <ProjectGuid>c1930979-c824-496b-a630-70f5369a636f</ProjectGuid>
```
Comment (cston) [inline src/Dependencies/PooledObjects/Microsoft.CodeAnalysis.PooledObjects.shproj]: Can be reverted.

**Comment by cston** [comment]
LGTM other than minor comments.

File: src/Dependencies/PooledObjects/Microsoft.CodeAnalysis.PooledObjects.shproj (hunk 4 above)
Comment (CyrusNajmabadi) [inline src/Dependencies/PooledObjects/Microsoft.CodeAnalysis.PooledObjects.shproj]: Looks like we have a misauthoring in our solution.  This is VS just trying to keep things in sync.  I've got htis change in all my PRs. Once it gets fixed up, it'll dissapear from all PRs.

File: src/Compilers/Core/Portable/Syntax/SyntaxNavigator.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/Syntax/SyntaxNavigator.cs]: done.

File: src/Compilers/Core/Portable/Syntax/SyntaxNavigator.cs (hunk 3 above)
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/Syntax/SyntaxNavigator.cs]: Done.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler Can anyone else take a look?

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler Need another set of eyes.  Thanks!

**Pull request merged**